backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java

```diff
@@ -1746,4 +1746,40 @@ public PaymentRule getDefaultPaymentRule()
 		return sysconfigs.getReferenceListAware(SYSCONFIG_C_Invoice_PaymentRule, PaymentRule.OnCredit, PaymentRule.class);
 	}
 
+	@Override
+	public final void discountInvoice(final org.compiere.model.I_C_Invoice invoice, final BigDecimal discountAmt, final String description)
```

Comment:
`discount` shall be Amount.
Pls avoid using BigDecimal for amounts and quantities.

Reply:
also document that `discountAmt` is not AP corrected.
i.e. "10 EUR discount" means "-10 EUR" in case of Sales Invoice and "+10 EUR" in case of Vendor invoice.